tests/vanilla/store.test.tsx

```diff
@@ -440,3 +440,162 @@ describe('async atom with subtle timing', () => {
     expect(await bValue).toBe(2)
   })
 })
+
+describe('unstable_resolve resolves the correct value for', () => {
+  function nextTask() {
+    return new Promise((resolve) => setTimeout(resolve))
+  }
+
+  it('primitive atom', async () => {
+    const store = createStore()
+    store.unstable_resolve = (atom: Atom<unknown>): Atom<unknown> => {
```

Comment:
Should it be this to pass the type test?
```suggestion
    store.unstable_resolve = <T>(atom: Atom<T>): Atom<T> => {
```

Reply:
But it could return any typed atom, right?

Reply:
Generally speaking, it should return the atom with the same capability, not just types.

Reply:
Ok, lets strongly type it then.